### PR TITLE
Rendering data of clicked product in comparison modal

### PR DESCRIPTION
**Pull request by jabrake**
- Added CSS styling for "related items" product cards
- Implemented toggle functionality to display comparison modal when clicking a product card
- Rendering data in comparison modal (passing state of clicked item card to modal via on click function)

@matthewwrobel can you review this one instead? going to close the older one, thanks!

### CODE REVIEW

File: client/related_items/ItemCard.jsx [hunk 1]
```diff
@@ -19,32 +20,37 @@ class ItemCard extends React.Component {
...
-            });
-          })
-        )
+        rating: '4.5',
+        features: productInfo.features
+      }));
+
+    helpers.getProductStyles(this.props.id)
+      .then(productStyles => helpers.findDefaultStyle(productStyles.results)
+        .then(defaultStyle => {
```
Comment (matthewwrobel) [inline client/related_items/ItemCard.jsx:29]: You can cut out one of these .then() calls if you make .findDefaultStyle a sync function (see my comment below...)

File: client/related_items/helpers.js [hunk 2]
```diff
@@ -28,14 +28,12 @@ const findDefaultStyle = (styles) => {
   return new Promise (resolve => {
```
Comment (matthewwrobel) [inline client/related_items/helpers.js:28]: Just a thought--You don't need to create Promises out of this synchronous helper function. You could make this sync and stick it inside of a .then block to cut down on some of your promise chaining.

File: client/related_items/ItemCard.jsx [hunk 3]
```diff
@@ -19,32 +20,37 @@ class ItemCard extends React.Component {
...
-              price: price
-            });
-          })
-        )
+        rating: '4.5',
+        features: productInfo.features
+      }));
+
+    helpers.getProductStyles(this.props.id)
+      .then(productStyles => helpers.findDefaultStyle(productStyles.results)
```
Comment (matthewwrobel) [inline client/related_items/ItemCard.jsx:28]: You can cut out one of these .then() calls by making findDefaultStyle() synchronous (see my comment below)

**Review by matthewwrobel — APPROVED** [review]
Looks really great, Jorge!

File: client/related_items/helpers.js (hunk 2 above)
Comment (jabrake) [reply client/related_items/helpers.js:28]: Ah really good catch! Will work on that for the next PR 👍

**Pull request merged**